src/test/lib/storage/mvcc_data_test.cpp

```diff
@@ -0,0 +1,132 @@
+#include "base_test.hpp"
+
+#include "storage/chunk.hpp"
+#include "storage/mvcc_data.hpp"
+
+namespace hyrise {
+
+class MvccDataTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    _mvcc_data = std::make_shared<MvccData>(ChunkOffset{3}, CommitID{1});
+    _mvcc_data->set_begin_cid(ChunkOffset{0}, CommitID{2});
+    _mvcc_data->set_end_cid(ChunkOffset{0}, CommitID{2});
+
+    _mvcc_data->set_begin_cid(ChunkOffset{1}, CommitID{3});
+    _mvcc_data->set_end_cid(ChunkOffset{1}, CommitID{4});
+    _mvcc_data->set_tid(ChunkOffset{1}, TransactionID{1});
+  }
+
+  std::shared_ptr<MvccData> _mvcc_data;
+};
+
+TEST_F(MvccDataTest, MemoryUsage) {
+  // _mvcc_data stores at least:
+  //   - 3 begin CIDs (uint32_t)
+  //   - 3 end CIDs (uint32_t)
+  //   - 3 TIDs (atomic uint32_t)
+  //   - max begin and end CIDs (atomic uint32_t)
+  //   - number of pending inserts (atimic uint32_t)
+  // Thus, the size should be larger than 12 * 8 B = 96 B.
+  EXPECT_GT(_mvcc_data->memory_usage(), 96);
```

Comment:
`GT` for not having to care about size added by atomics to data types, inlining, and sizes of the vector objects. `8` is genuinely wrong, should be `4`. Plus, we are not guaranteed that The vectors have an axact capacity of 3 by the standard (though libc++ and libstdc++ both do that.

Reply:
If we're using capacity, I totally understand the issue. But the function uses size.
But looking at `memory_usage()`, an exact check would be pretty much copying the code. `GT` still bother me, because arbitrarily large values are fine here.

Hmmm ...

Reply:
We use `capacity()` in the calculation 👀 
https://github.com/hyrise/hyrise/blob/d828141367c344bcce010376ae0a492ea964b3e2/src/lib/storage/mvcc_data.cpp#L77-L83

If you'd like to, I can also get the exact value here and compare w/ `EQ`. But then we have to admit this is a number we obtained empirically.

Reply:
Oh right, the PR changed it.